charts/milvus/values.yaml

```diff
@@ -739,3 +739,7 @@ externalPulsar:
 externalKafka:
   enabled: false
   brokerList: localhost:9092
+  sasl:
```

Comment:
Here I propose to use a different config structure like this:
```
externalKafka:
  enabled: false
  brokerList: localhost:9092
  mechanisms: plain
  securityProtocol: SASAL_SSL
  sasl:
     protocol: plain
     username: ""
     password: ""
```
How about this?

Reply:
Totally makes sense. I have applied your suggestion to this PR. 

As a side note, the reason I had the configuration as:

```
sasl:
  plain:
    ....
```

is because with mechanism configured, the parameters to set on the broker change. e.g. for PLAIN, `username` and `password` needs to be set. For something like `OAUTHBEARER `, the parameters to configure will be different from username and password.

But since this is missing on the milvus service side, we can come back to this later when other mechanisms are to be supported.